controllers/external/tracker_test.go

```diff
@@ -53,7 +54,7 @@ func newWatchCountController(raiseError bool) *watchCountController {
 	}
 }
 
-func (c *watchCountController) Watch(_ source.Source) error {
+func (c *watchCountController) Watch(_ source.TypedSource[reconcile.Request]) error {
```

Comment:
@alvaroaleman Wondering if there is a way to avoid this. Basically Source is the same as source.TypedSource[reconcile.Request], but the TypedController interface requires source.TypedSource[T]

Compile error without this change
```go
# sigs.k8s.io/cluster-api/controllers/external [sigs.k8s.io/cluster-api/controllers/external.test]
./tracker_test.go:67:39: cannot use ctrl (variable of type *watchCountController) as "sigs.k8s.io/controller-runtime/pkg/controller".TypedController[reconcile.Request] value in struct literal: *watchCountController does not implement "sigs.k8s.io/controller-runtime/pkg/controller".TypedController[reconcile.Request] (wrong type for method Watch)
		have Watch(source.Source) error
		want Watch(source.TypedSource[reconcile.Request]) error
```